File: experiments/runs/2026-09-20_06-59-55_claim-office-java-example-mapping_exact-ptdd-v1-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.util.Map;

/** The MHPCO price list: what each kind of item is insured for and what it costs to insure. */
final class PriceList {

    private record Entry(int insuranceValue, int basePremium) {
    }

    private static final Map<String, Entry> ENTRIES = Map.of(
            "sword", new Entry(1000, 100),
            "amulet", new Entry(600, 60),
            "staff", new Entry(800, 80),
            "potion", new Entry(400, 40),
            "rune", new Entry(250, 25),
            "moonstone", new Entry(250, 25));

    private PriceList() {
    }

    static boolean lists(String type) {
        return ENTRIES.containsKey(type);
    }

    static int insuranceValueOf(String type) {
        return entryFor(type).insuranceValue();
    }

    static int basePremiumOf(String type) {
        return entryFor(type).basePremium();
    }

    private static Entry entryFor(String type) {
        return ENTRIES.get(type);
    }
}

File: experiments/runs/2026-09-23_14-27-12_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking/src/price-list.ts
/** What the MHPCO insures: the price list prices it, the policy carries it. */
export interface Item {
  type: string;
  cursed?: boolean;
  enchantment?: number;
}

/**
 * One row of the MHPCO price list. The office publishes both columns for an item
 * type in one editorial act, so they are carried together rather than in separate
 * tables that would have to be kept aligned by hand.
 */
interface PriceListEntry {
  insuranceValue: number;
  basePremium: number;
}

/** Main items are listed by type, each row carrying both published columns. */
const MAIN_ITEMS: Record<string, PriceListEntry> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

/** Components are not listed by type: the office carries the whole class at one rate. */
const COMPONENT_TYPES = ["rune", "moonstone"];
const COMPONENT_ENTRY: PriceListEntry = { insuranceValue: 250, basePremium: 25 };

export function isComponent(item: Item): boolean {
  return COMPONENT_TYPES.includes(item.type);
}

/**
 * The price list is the single authority on what the MHPCO insures: an item type it
 * does not carry is uninsurable, so the lookup that prices an item is also the one
 * that rejects it. Both columns are refused together, because a type the office does
 * not list has neither a premium nor a value.
 */
function entryFor(item: Item): PriceListEntry {
  const entry = isComponent(item) ? COMPONENT_ENTRY : MAIN_ITEMS[item.type];
  if (entry === undefined) {
    throw new Error(`The MHPCO price list does not cover items of type "${item.type}".`);
  }
  return entry;
}

/** What the office charges to insure an item. */
export function basePremiumOf(item: Item): number {
  return entryFor(item).basePremium;
}

/** What the office carries for an item -- the figure a policy's cover is built on. */
export function insuranceValueOf(item: Item): number {
  return entryFor(item).insuranceValue;
}
